### PR TITLE
Update security checks

### PR DESCRIPTION
**Pull request by Zeitsperre**
## Overview

* Added custom endpoints for security in `main.yml`
* Now using commit hashes for specifying action versions

## Additional Information

I think `raven` will be ready for release after this PR

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ name: raven-wps Testing
 on:
   push:
     branches:
-      - master
+      - main
```
Comment (tlvu) [inline .github/workflows/main.yml:6]: When will this default branch name change be "live"?  You will take care to sync with the changes here https://github.com/bird-house/finch/issues/355#issuecomment-2096900447 to ensure the notebook autodeploy and nightly Jenkins is not broken?

**Review by tlvu — APPROVED** [review]
Looks good to me but please also ensure tuto notebook autodeploy to PAVICS and nightly Jenkins are not broken with this change.  Take a look at my comment on Finch repo for the same default branch name change.

File: .github/workflows/main.yml (hunk 1 above)
Comment (Zeitsperre) [reply .github/workflows/main.yml:6]: This has been live for Raven for some time now. I imagine that there aren't any problems.

Comment (tlvu) [reply .github/workflows/main.yml:6]: Ah true the branch switch has been already done for Raven, I forgot.  So this github CI workflow change was missing to complete the switch.

All good then.

**Pull request merged**